python3Packages.pylibgen: 2.0.1 -> 2.0.2

jonringer opened the pull request:
###### Motivation for this change
noticed it was broken while reviewing one of my other PRs

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @nico202 

```
[2 built, 0.0 MiB DL]
https://github.com/NixOS/nixpkgs/pull/66242
2 package were build:
papis python37Packages.pylibgen
```
```
[nix-shell:~/.cache/nix-review/pr-66242]$ ./results/papis/bin/papis --help
Usage: papis [OPTIONS] COMMAND [ARGS]...

Options:
  -h, --help                      Show this message and exit.
...
```

jonringer commented:
@GrahamcOfBorg build papis

worldofpeace commented on pkgs/development/python-modules/pylibgen/default.nix:
```diff
@@ -19,7 +21,7 @@ buildPythonPackage rec {
   checkInputs = [ pytest ];
 
-  # no tests in PyPI tarball
+  # too lazy to add pre-commit package
```
what does this mean, we have `gitAndTools.pre-commit` make you can use `toPythonModule`?

jonringer replied:
there's a seperate pypi package with the same name. I was trying to go home and just wanted to push the progress. I guess i didn't have to open a pr... xD

nico202 commented:
I'm moving from nix to guix, so if somebody wants to take the ownership of the package s/he's welcome

jonringer commented:
@nico202 unrelated to the PR, but I'm curious as to the motive behind moving to guix. Other than being able to use scheme and more of a focus on libre software, I'm not sure of a big difference.

nico202 commented:
Well the focus on the libre software is an important issue for me. Also, if at first I had difficulties in writing packages in scheme, now I find it to be a better language (after years I was able to have julia packages almost working, I was not able to do the same in nix). Another thing that at first I thought was better in nix was nix-shell, but guix environment is way more powerful. Then, I don't think guix pack (create distributable tarballs) has something similar in the nix world. Lastly, being an emacs user, the emacs-guix interface is superior to nox or anything else. I'm really happy with guix, even if there are way less packages, package management is a slower and to send patches you need emails :D

jonringer commented:
@nico202 thanks for the response. Wish you the best :)

worldofpeace commented:
Were you going to address https://github.com/NixOS/nixpkgs/pull/66242#discussion_r311595929 @jonringer?

jonringer commented:
yea, sorry, Keep getting distracted ;)

Commit:
pythonPackage.pre-commit: move to python-modules

  Add pkgs.gitAndTools.pre-commit alias
  Add pkgs.pre-commit alias

Commit:
python3Packages.pylibgen: 2.0.1 -> 2.0.2

jonringer commented:
@worldofpeace looks llike all the tests try to do a network call..... so it was all for not, trying to get tests working :rofl:

worldofpeace commented on pkgs/top-level/python-packages.nix:
```diff
@@ -1887,6 +1887,8 @@ in {
 
   pkginfo = callPackage ../development/python-modules/pkginfo { };
 
+  pre-commit = disabledIf isPy27 (toPythonModule (pkgs.gitAndTools.pre-commit));
```
I'm seeing support here for python 2.7 https://github.com/pre-commit/pre-commit/blob/v1.18.0/setup.cfg#L38

jonringer replied:
interesting... I was basing that off the gitAndTools.pre-commit package.....

jonringer replied:
I remember now, the application uses python3Packages, I can invert the logic..... and move pre-commit to being a python package, and then just do `toPythonApplication python3Packages.pre-commit` in the all-packages.nix.

But.... seeing as I don't have a reason to use it in this PR, i might remove the related changes altogether.

worldofpeace commented on pkgs/development/python-modules/pylibgen/default.nix:
```diff
@@ -1,31 +1,33 @@
   pname = "pylibgen";
-  version = "2.0.1";
+  version = "2.0.2";
+  disabled = !isPy3k;
```
This is a little more strict https://github.com/joshuarli/pylibgen/blob/master/setup.py#L44

jonringer replied:
hmm that's my bad, i stopped testing for 3.5, and 3.6 due to all the package builds :(

worldofpeace commented:
> @worldofpeace looks llike all the tests try to do a network call..... so it was all for not, trying to get tests working rofl

Aww. Sorry about that.

jonringer commented:
you couldn't have know, I couldn't have known :(

jonringer commented:
@GrahamcOfBorg build pythonPackages.pre-commit python3Packages.pre-commit python36Packages.pylibgen

jonringer commented:
moved the pre-commit package from being an application under git-and-tools, to a proper python packages (supports python{2.7,3.5,3.6,3.7} now), then used the "python3Packages.toPythonApplication" function to import it back under the git-and-tools namespace. Also added an alias for it under all-packages to the gitAndTools.pre-commit package.

pylibgen now builds with python3.{6,7}

jonringer commented:
@FRidh rebased off master, no more merge conflicts. Edited pre-commit message to now reflect that i moved the expression from gitAndTools to pythonPackages,
